src/main/kotlin/pt/up/fe/ni/website/backend/dto/entity/EntityDto.kt

```diff
@@ -57,9 +59,13 @@ abstract class EntityDto<T : Any> {
         // The use of suppress is explained at https://github.com/NIAEFEUP/website-niaefeup-backend/pull/20#discussion_r985236224
         @Suppress("UNCHECKED_CAST")
         private fun <T : Any> getTypeConversionClass(clazz: KClass<out EntityDto<T>>): KClass<T>? {
-            val thisType = clazz.supertypes.first { it.classifier == EntityDto::class }
+            val superType = clazz.supertypes.firstOrNull { it.jvmErasure.isSubclassOf(EntityDto::class) } ?: return null
+            val conversionClassArg =
+                superType.arguments.firstOrNull {
+                    it.type?.jvmErasure?.findAnnotation<Entity>() != null
+                } ?: return getTypeConversionClass(superType.jvmErasure as KClass<out EntityDto<T>>)
 
-            return thisType.arguments.firstOrNull()?.type?.jvmErasure as KClass<T>?
+            return conversionClassArg.type?.jvmErasure as KClass<T>?
```

Comment:
In order to convert Dtos to entities, `EntityDto` checks the dto's class definition to check the type definition (as in `ProjectDto() : EntityDto<Project>` we want to look for Project). Before this update, only direct subclasses of `EntityDto` used to work.

This change recursively looks for en entity as long as the dto is a subclass of `EntityDto`. This was needed because ProjectDto and EventDto now extend ActivityDto (`ProjectDto() : ActivityDto<Project>`)